File: backend/Domain/Entities/StorageArea.cs
namespace Domain.Entities
{
    public class StorageArea : BaseEntity<Guid>
    {
        public Guid StorageAreaTypeId { get; set; }
        public string Name { get; set; }
        public virtual StorageAreaType StorageAreaType { get; set; }
        public int TotalSpace { get; set; }
        public virtual ICollection<CustomerBox> CustomerBoxes { get; set; }

        public StorageArea()
        {
            CustomerBoxes = new HashSet<CustomerBox>();
        }
    }
}

File: frontend/front-desk-app/src/app/models/storageArea.ts
import { IStorageFacility } from "./storageFacility";
import { IStorageAreaType } from "./storageAreaType";
import { ICustomerBox } from "./customerBox";
import { IBaseEntity } from "./baseEntity";

export interface IStorageArea extends IBaseEntity<string> {
    storageFacilityId: string;
    storageFacility: IStorageFacility;
    storageAreaTypeId: string;
    name: string;
    storageAreaType: IStorageAreaType;
    totalSpace: number;
    customerBoxes: ICustomerBox[];
}